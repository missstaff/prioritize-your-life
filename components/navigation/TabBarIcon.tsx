import Ionicons from "@expo/vector-icons/Ionicons";
import { type IconProps } from "@expo/vector-icons/build/createIconSet";
import { type ComponentProps } from "react";

const TabBarIcon = ({
  style,
  ...rest
}: IconProps<ComponentProps<typeof Ionicons>["name"]>): JSX.Element => {
  return <Ionicons size={28} style={[{ marginBottom: -3 }, style]} {...rest} />;
};

export default TabBarIcon;
