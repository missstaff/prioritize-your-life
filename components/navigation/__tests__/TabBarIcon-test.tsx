import React from "react";
import { render } from "@testing-library/react-native";
import { TabBarIcon } from "../TabBarIcon";

jest.mock("@expo/vector-icons/Ionicons", () => "Ionicons");

describe("TabBarIcon", () => {
  afterEach(() => {
    jest.clearAllMocks();
  });

  it("renders correctly with the given props", () => {
    const { getByTestId } = render(
      <TabBarIcon name="home" testID="icon" style={{ color: "red" }} />
    );

    const icon = getByTestId("icon");

    expect(icon).toBeTruthy();
    expect(icon.props.style).toEqual([{ marginBottom: -3 }, { color: "red" }]);
    expect(icon.props.name).toBe("home");
  });
});
