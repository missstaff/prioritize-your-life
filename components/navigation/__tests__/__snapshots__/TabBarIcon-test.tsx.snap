// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TabBarIcon matches snapshot 1`] = `
<Ionicons
  name="home"
  size={28}
  style={
    [
      {
        "marginBottom": -3,
      },
      {
        "color": "red",
      },
    ]
  }
  testID="icon"
/>
`;
